tests/shell/tests.py

```diff
@@ -94,3 +95,30 @@ def test_shell_with_bpython_not_installed(self, select):
     # returns EOF and so select always shows that sys.stdin is ready to read.
     # This causes problems because of the call to select.select() toward the
     # end of shell's handle() method.
+
+    def test_ipython(self):
+        cmd = shell.Command()
+        mock_ipython = mock.Mock(start_ipython=mock.MagicMock())
+
+        with mock.patch.dict(sys.modules, {"IPython": mock_ipython}):
+            cmd.ipython(None)
+
+        assert mock_ipython.start_ipython.mock_calls == [mock.call(argv=[])]
+
+    def test_bpython(self):
+        cmd = shell.Command()
+        mock_bpython = mock.Mock(embed=mock.MagicMock())
+
+        with mock.patch.dict(sys.modules, {"bpython": mock_bpython}):
+            cmd.bpython(None)
```

Comment:
ditto on matching the type
```suggestion
            cmd.bpython({})
```